sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+            storeId: 'cbl-competencies-loaded',
+            model: 'Slate.cbl.model.Competency'
+        });
+    },
+
+
+    //event handlers
+    onComponentRender: function(dashboardView) {
+        var me = this,
+            siteEnv = window.SiteEnvironment || {},
+            student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
+            studentDashboardCompetenciesList = dashboardView.el.down('#studentDashboardCompetenciesList'),
+            contentArea = siteEnv.cblContentArea && Ext.create('Slate.cbl.model.ContentArea', siteEnv.cblContentArea),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            competenciesTpl = Ext.XTemplate.getTpl(me.view, 'competenciesTpl');
+
+        if (!student || !contentArea) {
+            return;
+        }
+
+        // empty competencies list
+        studentDashboardCompetenciesList.empty();
+        studentDashboardCompetenciesList.removeCls('competencies-unloaded').addCls('competencies-loading');
+
+        contentArea.getCompetenciesForStudents([student.getId()], function(competencies) {
+            var competenciesLength = competencies.length,
+                competencyIndex = 0,
+                competency, skillsList;
+            
+
+            if(!competenciesStore.isLoaded()) {
+                competenciesStore.loadRawData(competencies);
+            }
+
+            competenciesTpl.overwrite(studentDashboardCompetenciesList, {
+                student: student.getData(),
+                competencies: competencies
+            });
+
+            studentDashboardCompetenciesList.removeCls('competencies-loading').addCls('competencies-loaded');
+
+            for (; competencyIndex < competenciesLength; competencyIndex++) {
+                competency = Ext.create('Slate.cbl.model.Competency', competencies[competencyIndex]);
+                skillsList = studentDashboardCompetenciesList.down('.cbl-competency-panel[data-competency="'+competency.getId()+'"] .cbl-skill-meter');     
+                me.loadSkills(student, competency, skillsList);
+            }
+        });
+    },
+    
+    onDemoCellClick: function(dashboardView, ev, targetEl) {
+
+        Ext.create('Slate.cbl.view.student.skill.OverviewWindow', {
+            autoShow: true,
+            animateTarget: targetEl,
+
+            student: window.SiteEnvironment.cblStudent.ID,
+            competency: targetEl.up('ul.cbl-skill-demos').up('li.cbl-competency-panel').getAttribute('data-competency'),
+            skill: targetEl.getAttribute('data-skill')
+        });
+    },
+    
+
+
+    // protected methods
+    loadSkills: function(student, competency, skillsList) {
+        var me = this,
+            skillsTpl = Ext.XTemplate.getTpl(me.view, 'skillsTpl'),
+            skills, demonstrations, _renderSkills;
+
+        skillsList.removeCls('skills-unloaded').addCls('skills-loading');
+
+        _renderSkills = function() {
```

Comment:
Why isn't this defined as:
```javascript
function _renderSkills() {}
```
?

It will be private because it's a nested function, you don't even need the underscore.